tiledb/sm/c_api/tiledb.cc

```diff
@@ -7109,3 +7109,25 @@ int32_t tiledb_fragment_info_dump(
   fragment_info->fragment_info_->dump(out);
   return TILEDB_OK;
 }
+
+/* ********************************* */
+/*          EXPERIMENTAL APIs        */
+/* ********************************* */
+
+/** TODO */
+TILEDB_EXPORT int32_t tiledb_query_get_status_details(
+    tiledb_ctx_t* ctx,
+    tiledb_query_t* query,
+    tiledb_query_status_details_t* status) {
```

Comment:
Why add another function for query details? The signature of `tiledb_query_get_status` has an [output] argument `tiledb_query_status_t *status`. That's a pointer to an opaque-to-the-API type. It would seem to be better to add details to that type directly. You'd still need to add a function to obtain details, but it wouldn't be query-specific.

Reply:
`tiledb_query_status_t` is currently an enum, used by nearly all libtiledb downstream libraries.

Reply:
It's an enumeration, but is it opaque, or not? Are the values of the enumeration used directly, or not? If the values of the enumeration are only used internally to the library (say, to provide access to constant strings), then it's possible to change the type transparently.

If it's not an opaque type, then it's an API flaw. We may have to live with, but it would be blinkering ourselves to think it was a good idea.

Reply:
Yes, the enum values are used directly: https://github.com/search?q=org%3ATileDB-Inc+tiledb_incomplete&type=code